websocketOutportDriver/sender/webSocketSender.go

```diff
@@ -138,71 +133,58 @@ func (w *webSocketSender) start() {
 	}
 }
 
-func (w *webSocketSender) sendWithRetrial(data []byte, ackData []byte, counter uint64) {
-	clients := w.getClients()
+func (w *webSocketSender) sendDataToClients(
+	data []byte,
+	counter uint64,
+) error {
+	numSent := 0
+	var err error
+
+	clients := w.clientsHolder.GetAll()
 	if len(clients) == 0 {
-		w.log.Warn("no client to send to")
+		return ErrNoClientToSendTo
 	}
 
-	ticker := time.NewTicker(time.Millisecond * retrialTimeoutMS)
-	defer ticker.Stop()
-
-	for {
-		select {
-		case <-ticker.C:
-			clients = w.getClients()
-			if len(clients) == 0 {
-				continue
-			}
-			dataSent := w.sendDataToClients(data, ackData, clients, counter)
-			if dataSent {
-				return
-			}
+	for _, client := range w.clientsHolder.GetAll() {
+		err = w.sendData(data, *client, counter)
+		if err != nil {
+			w.log.Error("couldn't send data to client", "error", err)
+			continue
 		}
-	}
-}
 
-func (w *webSocketSender) sendDataToClients(
-	data []byte,
-	ackData []byte,
-	clients map[string]*webSocketClient,
-	counter uint64,
-) bool {
+		numSent++
+	}
 
-	for _, client := range clients {
-		result := w.sendData(data, ackData, *client, counter)
-		if !result {
-			return false
-		}
+	if numSent == 0 {
```

Comment:
debate-able: we might have a situation when the sender should mandatory send to all clients or fail. Might add a TODO here as to add an option to choose:
1. move on even if all clients failed to receive the message 
2. move on if at least one client received the message
3. move on if and only if all clients received the message

Reply:
yes, we should decide how to do these.